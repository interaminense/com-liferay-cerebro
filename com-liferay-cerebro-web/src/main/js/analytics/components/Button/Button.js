import JSXComponent from 'metal-jsx';

class Button extends JSXComponent {
	render() {
		return (
			<button
				class={`btn btn-dashboard btn-${this.props.style}`}
				type="button"
			>
				{this.props.label}
			</button>
		);
	}
}

export default Button;