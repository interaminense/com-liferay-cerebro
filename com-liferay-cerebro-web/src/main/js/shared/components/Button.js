import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

const DISPLAYS = [
	'danger',
	'dark',
	'info',
	'light',
	'link',
	'primary',
	'secondary',
	'success',
	'unstyled',
	'warning'
];

const SIZES = ['sm', 'lg'];

class Button extends Component {
	render() {
		const {
			active,
			block,
			display,
			monospaced,
			outline,
			size,
			type
		} = this.props;

		const classes = getCN(
			'button-root',
			'btn',
			`btn${outline ? '-outline' : ''}-${display}`,
			{
				active,
				['btn-block']: block,
				[`btn-${size}`]: size,
				'btn-monospaced': monospaced,
				'nav-link': display === 'link' && this.context.navBar
			}
		);

		return (
			<button {...this.otherProps()} class={classes} type={type}>
				{this.props.children}
			</button>
		);
	}
}

Button.PROPS = {
	active: Config.bool(),
	block: Config.bool(),
	display: Config.oneOf(DISPLAYS).value('secondary'),
	elementClasses: Config.string(),
	monospaced: Config.bool().value(false),
	outline: Config.bool(),
	size: Config.oneOf(SIZES),
	type: Config.string().value('button')
};

Button.DISPLAYS = DISPLAYS;
Button.SIZES = SIZES;

export default Button;