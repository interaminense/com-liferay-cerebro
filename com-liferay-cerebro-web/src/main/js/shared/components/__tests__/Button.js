import Button from '../Button';

describe('Button', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Button();

		expect(component).toMatchSnapshot();
	});

	it('should render as block', () => {
		component = new Button({
			block: true
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with a different type', () => {
		component = new Button({
			type: 'submit'
		});

		expect(component).toMatchSnapshot();
	});
});