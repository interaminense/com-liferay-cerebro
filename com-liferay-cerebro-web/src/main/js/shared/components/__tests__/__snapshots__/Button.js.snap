// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `<button class="button-root btn btn-secondary" type="button"></button>`;

exports[`Button should render as block 1`] = `<button class="button-root btn btn-secondary btn-block" type="button"></button>`;

exports[`Button should render with a different type 1`] = `<button class="button-root btn btn-secondary" type="submit"></button>`;
